frontend/src/components/RecordsList.tsx

```diff
@@ -10,20 +10,27 @@ import styles from "./records.module.scss";
 import "ag-grid-community/styles/ag-grid.css";
 import "ag-grid-community/styles/ag-theme-alpine.css";
 import "ag-grid-enterprise";
-import { ColDef, IServerSideGetRowsParams } from "ag-grid-community";
+import {
+  ColDef,
+  IServerSideDatasource,
+  IServerSideGetRowsParams,
+} from "ag-grid-community";
 import { DataName, useHookLazyGeneric } from "../shared/types";
 import SamplesList from "./SamplesList";
 import { Sample, SampleWhere } from "../generated/graphql";
-import { defaultColDef } from "../shared/helpers";
+import {
+  defaultColDef,
+  prepareSampleMetadataForAgGrid,
+} from "../shared/helpers";
 import { PatientIdsTriplet } from "../pages/patients/PatientsPage";
 import { ErrorMessage, LoadingSpinner, Toolbar } from "../shared/tableElements";
 
 interface IRecordsListProps {
   colDefs: ColDef[];
   dataName: DataName;
-  nodeName?: string;
   lazyRecordsQuery: typeof useHookLazyGeneric;
   lazyRecordsQueryAddlVariables?: Record<string, any>;
+  prepareDataForAgGrid?: (data: any) => any;
```

Comment:
This new prop is a function that takes the data returned by the GraphQL query and transforms it to be fed directly into the AG Grid table. Previously, we were doing this for Patients data inside its AG Grid column definitions.

This approach has two benefits: (1) matching how we handle samples data/views and (2) more performant.